File: Components/WinForms/CS/PivotGridMainDemo/Modules/MultipleTotals.cs
using System;
using System.Collections;
using System.ComponentModel;
using System.Drawing;
using System.Windows.Forms;
using DevExpress.XtraEditors.Controls;
using DevExpress.Data.PivotGrid;
using DevExpress.Utils;

namespace DevExpress.XtraPivotGrid.Demos.Modules {
	public partial class MultipleTotals : DevExpress.XtraPivotGrid.Demos.TutorialControl {
		public MultipleTotals() {
			// This call is required by the Windows Form Designer.
			CreateWaitDialog();
			InitializeComponent();

            TutorialInfo.WhatsThisCodeFile = "CS\\PivotGridMainDemo\\Modules\\MultipleTotals.cs";
            TutorialInfo.WhatsThisXMLFile = "DevExpress.XtraPivotGrid.Demos.CodeInfo.MultipleTotals.xml";

			// TODO: Add any initialization after the InitializeComponent call
		}
		public override PivotGridControl ViewOptionsControl { get { return pivotGridControl1; } }
		public override PivotGridControl ExportControl { get { return pivotGridControl1; } }
        //<pivotGridControl1>
		private void MultipleTotals_Load(object sender, System.EventArgs e) {
			pivotGridControl1.DataSource = GetNWindData("SalesPerson");
			fieldYear.Group.FilterValues.FilterType = PivotFilterType.Included;
			fieldYear.Group.FilterValues.Values.Add(2015);
			pivotGridControl1.ValueImages = CategoryImageList;
			InitCustomTotals();
		}
        
		void InitCustomTotals() {
			fieldCategoryName.TotalsVisibility = PivotTotalsVisibility.CustomTotals;
			fieldCategoryName.CustomTotals.Add(PivotSummaryType.Average);
			fieldCategoryName.CustomTotals.Add(PivotSummaryType.Sum);
			fieldCategoryName.CustomTotals.Add(PivotSummaryType.Max);
			fieldCategoryName.CustomTotals.Add(PivotSummaryType.Min);
			fieldCategoryName.CustomTotals.Add(PivotSummaryType.Count);
		}

		private void pivotGridControl1_FieldValueImageIndex(object sender, DevExpress.XtraPivotGrid.PivotFieldImageIndexEventArgs e) {
			if(e.Field == fieldCategoryName && Comparer.Equals(e.ValueType, PivotGridValueType.Value))
				e.ImageIndex = GetCategoryIndexByName(e.Value);
			if(e.ValueType == PivotGridValueType.CustomTotal) 
				if(e.CustomTotal.SummaryType == PivotSummaryType.Sum && e.Field.Area != PivotArea.DataArea)
					e.ImageIndex = 8;
		}
        //</pivotGridControl1>
	}
}

File: Components/WinForms/VB/PivotGridMainDemo/Modules/MultipleTotals.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraEditors.Controls
Imports DevExpress.Data.PivotGrid
Imports DevExpress.Utils

Namespace DevExpress.XtraPivotGrid.Demos.Modules
	Partial Public Class MultipleTotals
		Inherits DevExpress.XtraPivotGrid.Demos.TutorialControl
		Public Sub New()
			' This call is required by the Windows Form Designer.
			CreateWaitDialog()
			InitializeComponent()

            TutorialInfo.WhatsThisCodeFile = "VB\PivotGridMainDemo\Modules\MultipleTotals.vb"
            TutorialInfo.WhatsThisXMLFile = "MultipleTotals.xml"

			' TODO: Add any initialization after the InitializeComponent call
		End Sub
		Public Overrides ReadOnly Property ViewOptionsControl() As PivotGridControl
			Get
				Return pivotGridControl1
			End Get
		End Property
		Public Overrides ReadOnly Property ExportControl() As PivotGridControl
			Get
				Return pivotGridControl1
			End Get
		End Property
		'<pivotGridControl1>
		Private Sub MultipleTotals_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles MyBase.Load
			pivotGridControl1.DataSource = GetNWindData("SalesPerson")
			fieldYear.Group.FilterValues.FilterType = PivotFilterType.Included
			fieldYear.Group.FilterValues.Values.Add(1995)
			pivotGridControl1.ValueImages = CategoryImageList
			InitCustomTotals()
		End Sub

		Private Sub InitCustomTotals()
			fieldCategoryName.TotalsVisibility = PivotTotalsVisibility.CustomTotals
			fieldCategoryName.CustomTotals.Add(PivotSummaryType.Average)
			fieldCategoryName.CustomTotals.Add(PivotSummaryType.Sum)
			fieldCategoryName.CustomTotals.Add(PivotSummaryType.Max)
			fieldCategoryName.CustomTotals.Add(PivotSummaryType.Min)
			fieldCategoryName.CustomTotals.Add(PivotSummaryType.Count)
		End Sub

		Private Sub pivotGridControl1_FieldValueImageIndex(ByVal sender As Object, ByVal e As DevExpress.XtraPivotGrid.PivotFieldImageIndexEventArgs) Handles pivotGridControl1.FieldValueImageIndex
			If e.Field Is fieldCategoryName AndAlso Comparer.Equals(e.ValueType, PivotGridValueType.Value) Then
				e.ImageIndex = GetCategoryIndexByName(e.Value)
			End If
			If e.ValueType = PivotGridValueType.CustomTotal Then
				If e.CustomTotal.SummaryType = PivotSummaryType.Sum AndAlso e.Field.Area <> PivotArea.DataArea Then
					e.ImageIndex = 8
				End If
			End If
		End Sub
		'</pivotGridControl1>
	End Class
End Namespace

